test/performance/scale_revision_by_load_test.go

```diff
@@ -180,10 +184,12 @@ func scaleRevisionByLoad(t *testing.T, numClients int) []junit.TestCase {
 	tc = append(tc, perf.CreatePerfTestCase(float32(resp.Result[0].ActualQPS), "actualQPS", t.Name()))
 	tc = append(tc, perf.CreatePerfTestCase(float32(resp.ErrorsPercentage(0)), "errorsPercentage", t.Name()))
 
-	for ev := range scaleCh {
+	m.Lock()
+	for _, ev := range scaleEvents {
 		t.Logf("Scaled: %d -> %d in %v", ev.oldScale, ev.newScale, ev.timestamp.Sub(resp.Result[0].StartTime))
 		tc = append(tc, perf.CreatePerfTestCase(float32(ev.timestamp.Sub(resp.Result[0].StartTime)/time.Second), fmt.Sprintf("scale-from-%02d-to-%02d(seconds)", ev.oldScale, ev.newScale), t.Name()))
 	}
+	m.Unlock()
```

Comment:
Let's defer this Unlock. We don't care really to Unlock it early and deferring is always safer.